dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -90,9 +89,23 @@
 
     private ApacheHttpClientChannels() {}
 
-    @VisibleForTesting
-    static Channel create(ClientConfiguration conf) {
-        String channelName = "apache-channel";
+    /**
+     * Simple cjr-style factory method. Prefer using {@code dialogue-clients} general purpose factory where possible.
+     * This method exists to ease of migration from hc4 to hc5.
+     *
+     * @deprecated Prefer dialogue-clients if possible, otherwise please provide a channel name for more accurate
+     * metrics using {@link #create(ClientConfiguration, String)}.
+     */
+    @Deprecated
+    public static Channel create(ClientConfiguration conf) {
+        return create(conf, "apache-channel");
+    }
+
+    /**
+     * Simple cjr-style factory method. Prefer using {@code dialogue-clients} general purpose factory where possible.
+     * This method exists to ease of migration from hc4 to hc5.
+     */
+    public static Channel create(ClientConfiguration conf, String channelName) {
```

Comment:
Added these two methods to facilitate moving existing code from hc4 to hc5 -- there are still a few lingering consumers and I don't want to cause problems for them.

Reply:
Maybe we can use this as an opportunity to push existing consumers to use fancy client factory interfaces that we expose instead?

Reply:
The blocker is the ServiceConfiguration API requiring a path to key material where we only have a SSLSocketFactory/trustManager in some cases. That's why we haven't been able to move the JaxrsClient.create code to the fancy factory yet :-(